src/sagemaker/cli/main.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import absolute_import
+
+import argparse
+import logging
+import sys
+
+import sagemaker
+
+logger = logging.getLogger(__name__)
+
+
+def parse_arguments(args):
+    # common arguments
+    common_parser = argparse.ArgumentParser(add_help=False)
+
+    # image-related settings
+    image_mtx = common_parser.add_mutually_exclusive_group(required=True)
+    image_mtx.add_argument('--tf', help='use a TensorFlow container image', action='store_true')
+    image_mtx.add_argument('--mx', help='use an MXNet container image', action='store_true')
+
+    # path to data and script files
+    common_parser.add_argument('--data', help='path to training data or model files', type=str, default='./data')
```

Comment:
one of the main goals for this cli is to enable easy iteration, so sensible defaults seems like the right thing to do.

Reply:
What's the cost relatively of typing out the extra parameters compared to the cost of understanding how to write the command initially? I think it's common to start out only specifying the parameters that are marked as required, and currently the error message isn't very helpful if those files don't exist. Can we at least improve those error messages?

The other issue with defaulting is hardware - we made the choice explicitly to not default those in the Python SDK to reduce the chances of over-charging or under-provisioning, since there's no one size fits all for all problems. What makes that not apply to this CLI?

Reply:
@tomfaulhaber i think you were party to the initial discussion about defaulting. how do you feel about it in this context?

Reply:
I think count is obvious: 1 (and we should do that in the SDK also - I disagree with not defaulting it)

Instance type is trickier, but I do feel like we should default it. I would go for ml.c4.xlarge here. Again, if it were up to me, I would add this default in the SDK, though I admit that it's way less obvious than the instance count.

Reply:
@jesterhazy pointed out that the ml.m4.xlarge are free tier eligible, so we should go with that